wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1748,3 +1756,54 @@ fn supports_format(
         _ => false,
     }
 }
+
+fn supports_bgra8unorm_storage(
+    instance: &ash::Instance,
+    phd: vk::PhysicalDevice,
+    api_version: u32,
+) -> bool {
+    // See https://github.com/KhronosGroup/Vulkan-Docs/issues/2027#issuecomment-1380608011
+
+    // We must query via FormatProperties3 only for vk 1.3 or later.
+    if api_version < vk::API_VERSION_1_3 {
+        // Note we might be able to get away with not checking `STORAGE_WRITE_WITHOUT_FORMAT_KHR`
+        // on older drivers, but for now we rely on it.
+        return false;
+    }
+
+    unsafe {
+        let mut properties3 = vk::FormatProperties3 {
+            s_type: vk::StructureType::FORMAT_PROPERTIES_3,
+            p_next: std::ptr::null_mut(),
+            linear_tiling_features: vk::FormatFeatureFlags2::empty(),
+            optimal_tiling_features: vk::FormatFeatureFlags2::empty(),
+            buffer_features: vk::FormatFeatureFlags2::empty(),
+        };
+
+        let p_next: *mut ash::vk::FormatProperties3 = &mut properties3;
+        let mut properties2 = vk::FormatProperties2 {
+            s_type: vk::StructureType::FORMAT_PROPERTIES_2,
+            p_next: p_next as *mut _,
+            format_properties: vk::FormatProperties {
+                linear_tiling_features: vk::FormatFeatureFlags::empty(),
+                optimal_tiling_features: vk::FormatFeatureFlags::empty(),
+                buffer_features: vk::FormatFeatureFlags::empty(),
+            },
+        };
```

Comment:
```suggestion
        let mut properties3 = vk::FormatProperties3::default();
        let mut properties2 = vk::FormatProperties2::builder().push_next(&mut properties3);
```

Noticed ash has some nice utils we can use here. Note that the builder derefs to the type it's building.

Reply:
Nice!